solutions/instances/provider.tf

```diff
@@ -0,0 +1,35 @@
+########################################################################################################################
+# Provider config
+########################################################################################################################
+
+provider "ibm" {
+  ibmcloud_api_key = var.ibmcloud_api_key
+}
+
+locals {
+  at_endpoint = "https://api.${var.region}.logging.cloud.ibm.com" # Todo: change it to private(api.private)
```

Comment:
you can determine whether to use public or private endpoint here based on the value passed for the `log_analysis_service_endpoints` variable